src/xopen/__init__.py

```diff
@@ -333,18 +333,28 @@ def _open_gz(filename, mode, compresslevel, threads):
         exc = FileNotFoundError  # was introduced in Python 3.3
     else:
         exc = OSError
+
     if 'r' in mode:
-        try:
+        def with_threads():
             return PipedGzipReader(filename, mode, threads=threads)
-        except exc:
-            # pigz is not installed
+
+        def without_threads():
             return buffered_reader(gzip.open(filename, mode))
     else:
-        try:
+        def with_threads():
```

Comment:
These functions obscure what is happening and diminish readability. It seems better to me to use the return statements directly in the `if threads==0` block.

Reply:
I want to avoid code duplication: The `except` and the `threads==0` cases are the same. The only way I can see to not have to duplicate the code is to move it into a function.

Reply:
Even after considering this for a while, I think this is more readable than the alternative (also because there’s now only a single try/except). I’ve changed the function names a bit, however, to improve readability.